### PR TITLE
libwebp-base 1.3.2 [rebuild for cmake]

### PR DESCRIPTION
**Pull request by ifitchet**
libwebp-base 1.3.2 

**Destination channel:** Defaults

### Links

- [PKG-5894]
- dev_url:        https://chromium.googlesource.com/webm/libwebp/tree/v1.3.2
- conda_forge:    https://github.com/conda-forge/libwebp-base-feedstock
- pypi:           n/a
- pypi inspector: n/a

### Explanation of changes:

- new build number
- as per conda-forge: swap autotools for cmake on all platforms
- this was great apart from `osx` where, after much head-scratching, we want to be setting `CMAKE_BUILD_WITH_INSTALL_RPATH` to avoid overlinking errors

### Effect:

- we don't get manual pages anymore (I think this is the first time I've seen manual pages so no great loss)
- there are some differences in the `.pc` files:

```
diff -r libwebp-base-1.3.2-h61381ae_0/lib/pkgconfig/libsharpyuv.pc libwebp-base-1.3.2-h61381ae_1/lib/pkgconfig/libsharpyuv.pc
1c1
< prefix=/opt/conda/conda-bld/libwebp-base_1728467173068/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_
---
> prefix=/opt/conda/conda-bld/libwebp-base_1728467227750/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_
11c11
< Libs.private: -lm -pthread 
---
> Libs.private: -lm  -lpthread
```

ie. the cmake build uses `-lpthread` whereas the autotools build uses `-pthread`.

- no library filename differences
- `SONAME`s look the same:

```
(gcc) bash-4.2# for x in {0,1} ; do for l in libwebp-base-1.3.2-h61381ae_$x/lib/*.so ; do aarch64-conda-linux-gnu-objdump -p $l ; done ; done | grep SONAME
  SONAME               libsharpyuv.so.0
  SONAME               libwebpdecoder.so.3
  SONAME               libwebpdemux.so.2
  SONAME               libwebpmux.so.3
  SONAME               libwebp.so.7
  SONAME               libsharpyuv.so.0
  SONAME               libwebpdecoder.so.3
  SONAME               libwebpdemux.so.2
  SONAME               libwebpmux.so.3
  SONAME               libwebp.so.7
```

- I've tried using it with dependees (`libwebp`, `opencv`) which seem to display help and report sensible things from `ldd` but we seem to be a bit short on more thorough testing examples

[PKG-5894]: https://anaconda.atlassian.net/browse/PKG-5894?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,19 +10,16 @@ source:
...
 requirements:
   build:
     - {{ compiler('c') }}
-    - cmake     # [win]
-    - libtool   # [not win]
-    - m2-patch  # [win]
-    - make      # [not win]
-    - ninja     # [win]
+    - cmake
+    - ninja
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - ninja-base
```

Comment (ifitchet) [reply recipe/meta.yaml]: That's seems reasonable.
It doesn't change the prefect builds of `osx-*`, though.

File: recipe/bld.bat [hunk 2]
```diff
@@ -1,33 +1,25 @@
...
+cd build
 
-:: Isolate the build.
-mkdir Build-%PKG_NAME%
-cd Build-%PKG_NAME%
-if errorlevel 1 exit /b 1
-
-:: Generate the build files.
-echo "Generating the build files..."
-cmake .. %CMAKE_ARGS% ^
```
Comment (JeanChristopheMorinPerso) [inline recipe/bld.bat:13]: Keep `%CMAKE_ARGS%` please

**Comment by JeanChristopheMorinPerso** [comment]
@ifitchet man pages are suposed to be built and installed according to https://chromium.googlesource.com/webm/libwebp/+/refs/tags/v1.3.2/CMakeLists.txt#775...

File: recipe/bld.bat (hunk 2 above)
Comment (ifitchet) [reply recipe/bld.bat:13]: added in commit below.

**Comment by ifitchet** [comment]
> @ifitchet man pages are suposed to be built and installed according to https://chromium.googlesource.com/webm/libwebp/+/refs/tags/v1.3.2/CMakeLists.txt#775...

Ah, they are built if the `WEBP_BUILD_${EXEC_BUILD}` flag is set which, given we copied the CF recipe, largely is not: https://github.com/conda-forge/libwebp-base-feedstock/blob/main/recipe/build.sh#L8-L17

**Review by skupr-anaconda — APPROVED** [review]

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**